src/Utils.cc

```diff
@@ -563,4 +574,271 @@ thread_type_t fbgemmGetThreadPartition(
   return th_info;
 }
 
+namespace {
+
+// implementation taken from pytorch/c10/util/llvmMathExtras.h
+template <typename T>
+size_t count_leading_zeros(T val) {
+  if (!val)
+    return std::numeric_limits<T>::digits;
+
+  size_t zero_bits = 0;
+  for (T shift = std::numeric_limits<T>::digits >> 1; shift; shift >>= 1) {
+    T tmp = val >> shift;
+    if (tmp)
+      val = tmp;
+    else
+      zero_bits |= shift;
+  }
+  return zero_bits;
+}
+
+// histogram size per thread
+constexpr int RDX_HIST_SIZE = 256;
+
+#define COMBINE_PREFIX_SUM_IN_RANGE(                            \
+    sum, prev_sum, bins_beg, bins_end, hist, hist_ps, nthreads) \
+  for (int bins = bins_beg; bins < bins_end; ++bins) {          \
+    for (int t = 0; t < nthreads; ++t) {                        \
+      sum += histogram[t * RDX_HIST_SIZE + bins];               \
+      histogram_ps[t * RDX_HIST_SIZE + bins] = prev_sum;        \
+      prev_sum = sum;                                           \
+    }                                                           \
+  }
+
+void combine_prefix_sum(
+    int nthreads,
+    int elements_count,
+    int* histogram,
+    int* histogram_ps) {
```

Comment:
By commenting out `elements_count` here we avoid the need for `(void)elements_count` later. Use `const` for safety+readability.
```suggestion
    const int nthreads,
    const int /*elements_count*/,
    int* const histogram,
    int* const histogram_ps) {
```

Reply:
I cannot comment out `elements_count`, as in the function body it is used in assert. The more important question is whether we need this assertion. I think it was used for precaution purposes, as we have `radix_sort` tests, do you think we still need it?